### PR TITLE
Failure handling: timeout

### PR DESCRIPTION
**Pull request by gluke77**
Partially addresses #80

---
This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-appcontroller/199)

### CODE REVIEW

**Comment by pigmej** [comment]
looks simple, can we have tests ?

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-:-Kd0g5khvURFz4D4WY6r:bofpas3)*

**Comment by gluke77** [comment]
sure, tests are in progress

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-:-Kd0mftmyZ89ybsuuGPw:bm8wilr)*

**Comment by gluke77** [comment]
@pigmej  so we have tests :)

**Comment by jellonek** [comment]
:lgtm:

---

Reviewed 9 of 9 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-:-KdjirzNNg0Br4ErK3Ep:bnfp4nl)*

File: e2e/basic_test.go [hunk 1]
```diff
@@ -173,6 +174,25 @@ var _ = Describe("Basic Suite", func() {
...
+			parentPod = DelayedPod("parent-pod", 15)
+			childPod = PodPause("child-pod")
+		})
+
+		Context("If parent timed out", func() {
+			It("on-error dependency must be followed", func() {
+				parentResDef := framework.WrapWithMetaAndCreate(parentPod, map[string]interface{}{"timeout": 5})
+				framework.ConnectWithMeta(parentResDef, framework.WrapAndCreate(childPod), map[string]string{"on-error": "true"})
+				framework.Run()
+				testutils.WaitForPod(clientset, namespace.Name, childPod.Name, v1.PodRunning)
```
Comment (dshulyak) [inline e2e/basic_test.go]: framework.Clientset

framework.Namespace.Name

**Comment by nebril** [comment]
:lgtm:

---

Reviewed 9 of 9 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[e2e/basic_test.go, line 192 at r2](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-Kd2tn3HQ7uUZyGEnCuU-r2-192:-Kdk9np-d4duxRFhc7Uz:bbsv7qx) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/d6ebed00339d669ab6f490e39adbb0a3515d6f73/e2e/basic_test.go#L192)):*
<details><summary><i>Previously, dshulyak (Dmitry Shulyak) wrote…</i></summary><blockquote>

framework.Clientset

framework.Namespace.Name
</blockquote></details>

All above tests use local variables, I guess it all needs changing (unless it's changed in your PR)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-:-KdkD16fbneYYdKqCF4I:bnfp4nl)*

**Comment by gluke77** [comment]
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[e2e/basic_test.go, line 192 at r2](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-Kd2tn3HQ7uUZyGEnCuU-r2-192:-Kdl9XB4wSSg0BWxib8o:bs0mnn8) ([raw file](https://github.com/mirantis/k8s-appcontroller/blob/d6ebed00339d669ab6f490e39adbb0a3515d6f73/e2e/basic_test.go#L192)):*
<details><summary><i>Previously, nebril (Maciej Kwiek) wrote…</i></summary><blockquote>

All above tests use local variables, I guess it all needs changing (unless it's changed in your PR)
</blockquote></details>

@dshulyak done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199)*

**Comment by gluke77** [comment]
Rebased against the latest master to avoid merge conflict

---

Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-:-Kdl9gt7M6Y-EB7E6IDe:b-c0qpe6)*

**Comment by nebril** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-:-Kdz-AEG_dKDT9hx0ol6:bnfp4nl)*

**Comment by pigmej** [comment]
:lgtm:

---

Reviewed 8 of 9 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/199#-:-KdzgCr4nWUPezbCk3x7:bnfp4nl)*

**Pull request merged**